core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -642,7 +642,7 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
 
   // For testing.
   private[history] def isFsInSafeMode(dfs: DistributedFileSystem): Boolean = {
-    dfs.setSafeMode(HdfsConstants.SafeModeAction.SAFEMODE_GET)
+    dfs.setSafeMode(HdfsConstants.SafeModeAction.SAFEMODE_GET, true)
```

Comment:
Should mention what the true is for (e.g. with a named param or just something like `/* isChecked - run on ActiveNN */`